trailblazer/server/api.py

```diff
@@ -68,7 +68,7 @@ def analyses():
     for analysis in query_page.all():
         analysis_data = analysis.to_dict()
         analysis_data["user"] = analysis.user.to_dict() if analysis.user else None
-        failed_job: Job | None = analysis.last_failed_job
+        failed_job: Job | None = store.get_latest_failed_job_for_analysis(analysis.id)
```

Comment:
This is not super efficient (running one extra query per analysis) but way better than the previous implementation. Created an issue for us to improve it https://github.com/Clinical-Genomics/trailblazer/issues/324